numpy/testing/nosetester.py

```diff
@@ -344,11 +325,17 @@ def test(self, label='fast', verbose=1, extra_argv=None,
              <http://nedbatchelder.com/code/modules/coverage.html>`_).
         raise_warnings : str or sequence of warnings, optional
             This specifies which warnings to configure as 'raise' instead
-            of 'warn' during the test execution.  Valid strings are:
+            of 'always' during the test execution.  Valid strings are:
 
-              - "develop" : equals ``(DeprecationWarning, RuntimeWarning)``
+              - "develop" : equals ``(Warning,)``
               - "release" : equals ``()``, don't raise on any warnings.
 
+        check_filters : None
+            If True, it will raise an error if the filter list changed during
+            the tests. This should not happen, since all tests should be
+            guarded by a ``catch_warnings`` or similar context manager.
+            If not given, will check if raise_warnings is in "develop" mode.
```

Comment:
This is an API change, and I am not sure if it is a good idea in any case, though it is probably the easiest way to check if someone got a warnings modification outside a context by accident.